hedera-node/src/test/java/com/hedera/services/txns/ethereum/EthTxDataTest.java

```diff
@@ -176,18 +180,66 @@ void whiteBoxDecodingErrors() {
 				oneByte, oneByte, oneByte, oneByte,
 				oneByte, oneByte, oneByte, oneByte,
 				oneByte);
+		var size_1 = List.of(
+				oneByte);
 
 		// legacy TX with too many RLP entries
 		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeAsList(size_13)));
+		// legacy TX with too few RLP entries
+		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeAsList(size_1)));
 		// type 2 TX with too many RLP entries
 		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeSequentially(new byte[] { 2 }, size_13)));
+		// type 2 TX with too few RLP entries
+		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeSequentially(new byte[] { 2 }, size_1)));
 		// Unsupported Transaciton Type
 		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeSequentially(new byte[] { 127 }, size_13)));
+		// Trimmed End Bytes
+		assertThrows(IllegalArgumentException.class, () ->
+				EthTxData.populateEthTxData(RLPEncoder.encodeSequentially(Hex.decode(RAW_TX_TYPE_0_TRIMMED_LAST_BYTES))));
 
 		// poorly wrapped typed transaction
 		assertNull(EthTxData.populateEthTxData(RLPEncoder.encodeSequentially(new byte[] { 2 }, oneByte, oneByte)));
 	}
 
+	byte[][] normalRlpData() {
+		var oneByte = new byte[] { 1 };
+		byte[][] rlpArray = new byte[][] {
+				oneByte,oneByte,oneByte,oneByte,
+				oneByte,oneByte,oneByte,oneByte,
+				oneByte,oneByte,oneByte,oneByte
+		};
+		return rlpArray;
+	}
+
+	@Test
+	void parsingErrors() {
+		final var wrongData = Hex.encode(ByteString.copyFromUtf8("wrong").toByteArray());
+
+		// invalid nonce
+		var normalData = normalRlpData();
+		normalData[1] = wrongData;
+		final var invalidNonceData = Arrays.asList(normalData);
+
+		assertThrows(IllegalArgumentException.class, () ->
```

Comment:
There we will have sonar code smell: Only one method invocation is expected when testing runtime exceptions

Reply:
Again, populateEthTxData should throw no exceptions, this should be assertNull